src/sample/environment/image_transforms.py

```diff
@@ -0,0 +1,127 @@
+import math
+from collections.abc import Callable
+from typing import override
+
+import torch
+import torch.nn as nn
+import torchvision.transforms.v2.functional as F
+from torch import Tensor
+from torchvision.transforms.v2 import (
+    Compose,
+    ToDtype,
+    ToImage,
+    ToPureTensor,
+)
+
+from pamiq_vrchat.sensors.image import ImageFrame
+
+
+class ResizeAndCenterCrop(nn.Module):
+    """Resize and center crop transform for images.
+
+    This module resizes the input image to fit within the target size
+    while maintaining aspect ratio, then performs a center crop to the
+    exact target size.
+    """
+
+    def __init__(self, size: tuple[int, int]) -> None:
+        """Initialize the ResizeAndCenterCrop transform.
+
+        Args:
+            size: Target size as (height, width) tuple.
+        """
+        super().__init__()
+        self.size = list(size)
+
+    @override
+    def forward(self, input: Tensor) -> Tensor:
+        """Apply resize and center crop to the input tensor.
+
+        Args:
+            input: Input tensor with shape (..., H, W) where H, W are height and width.
+
+        Returns:
+            Transformed tensor with shape (..., size[0], size[1]).
+
+        Raises:
+            ValueError: If input has less than 3 dimensions.
+            ValueError: If input height or width is 0.
+        """
+        if input.ndim < 3:
+            raise ValueError(
+                f"Input tensor must have at least 3 dimensions, got {input.ndim}"
+            )
+
+        inpt_img_size = input.shape[-2:]
+        if min(inpt_img_size) == 0:
+            raise ValueError(
+                f"Input image dimensions must be non-zero, got {tuple(inpt_img_size)}"
+            )
+
+        max_size = math.ceil(max(inpt_img_size) / min(inpt_img_size) * max(self.size))
+        input = F.resize(input, None, max_size=max_size)
```

Comment:
最小の辺の長さを最大のサイズ辺長さにスケールするのは一般に過剰に小さい範囲をくりぬくことになり、例えば(1,2)のインプットに対して(1,2)のサイズでくりぬくと(2,4)にリサイズされた上で中央部分がくりぬかれることになります。
以下を参考にアスペクト比ベースで直してみてください
```suggestion
        ar_inpt = inpt_img_size[1] / inpt_img_size[0]
        ar_size = self.size[1] / self.size[0]
        if ar_inpt < ar_size:
            scale_size = (math.ceil(self.size[0] * ar_inpt), self.size[1])
        else:
            scale_size = (self.size[0], math.ceil(self.size[1] / ar_inpt)
        input = F.resize(input, scale_size)
```

Reply:
修正しました。
https://github.com/MLShukai/pamiq-vrchat/pull/49/commits/2cadef6c6d754fd3177c1f9fca78169e2f0ed169